src/api/index.spec.tsx

```diff
@@ -0,0 +1,144 @@
+import { HttpMethod, callApi, logOutFromServer } from '.';
+import {
+  initialState as defaultApiState,
+  actions as apiActions,
+} from '../reducers/api';
+import configureStore from '../configureStore';
+
+describe(__filename, () => {
+  const getApiState = ({ authToken = '12345' } = {}) => {
+    const store = configureStore();
+
+    store.dispatch(apiActions.setAuthToken({ authToken }));
+    const { api } = store.getState();
+
+    return api;
+  };
+
+  describe('callApi', () => {
+    const callApiWithDefaultApiState = (params = {}) => {
+      return callApi({ apiState: defaultApiState, endpoint: '/', ...params });
+    };
+
+    it('calls the API', async () => {
+      await callApiWithDefaultApiState({ endpoint: '/foo/' });
+
+      expect(fetch).toHaveBeenCalledWith(`/api/v4/foo/`, {
+        headers: {},
+        method: 'GET',
+      });
+    });
+
+    it('adds a trailing slash to the endpoint if there is none', async () => {
+      await callApiWithDefaultApiState({ endpoint: '/foo' });
+
+      expect(fetch).toHaveBeenCalledWith(`/api/v4/foo/`, expect.any(Object));
```

Comment:
Maybe a helper for `toHaveBeenCalledWithURL`, which then checks the URL and does the `expect.any(Object)` thing would be nice. That would replace this more complex code in a number of the following tests.

Another option would be to create an object with the expected default values, e.g., `{ headers: {}, method: 'GET' }` in this case, and then pass that object in as the second argument to `toHaveBeenCalledWith`, and use the spread operator if you need to override one of those properties for a specific test.

I'm not sure how keen I am about having this `expect.any(Object)` in the tests.